palace/drivers/basesolver.cpp

```diff
@@ -235,13 +126,12 @@ void BaseSolver::SolveEstimateMarkRefine(
   // Helper to save off postprocess data.
   auto SavePostProcess = [&](int iter)
   {
-    Mpi::Barrier(comm);
     BlockTimer bt_io(Timer::IO);
+    Mpi::Barrier(comm);
```

Comment:
I think this isn't necessary? If anything it should be before the timer, but I think it makes sense for the timer to include all time waiting.

Reply:
I had put it there as I was worried about the root starting to move the post processing directory while another processor is still writing into it. Given that, I viewed that blocking as part of the IO, hence placing it within the barrier.